src/sys/unix.rs

```diff
@@ -785,8 +786,16 @@ fn recvmsg(
     msg.msg_namelen = msg_namelen;
     msg.msg_iov = bufs.as_mut_ptr().cast();
     msg.msg_iovlen = min(bufs.len(), IovLen::MAX as usize) as IovLen;
-    syscall!(recvmsg(fd, &mut msg, flags))
-        .map(|n| (n as usize, msg.msg_namelen, RecvFlags(msg.msg_flags)))
+    msg.msg_control = control_data.as_mut_ptr().cast();
+    msg.msg_controllen = control_data.len();
```

Comment:
At least on macOS this is a `u32`.
```suggestion
    msg.msg_controllen = control_data.len() as _;
```